docs/source/using-executorch-export.md

```diff
@@ -1,3 +1,173 @@
+
+```python
+import torch
+
+class Model(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+        self.seq = torch.nn.Sequential(
+            torch.nn.Conv2d(1, 8, 3),
+            torch.nn.ReLU(),
+            torch.nn.Conv2d(8, 16, 3),
+            torch.nn.ReLU(),
+            torch.nn.AdaptiveAvgPool2d([[1,1]])
+        )
+        self.linear = torch.nn.Linear(16, 10)
+    
+    def forward(self, x):
+        y = self.seq(x)
+        y = torch.flatten(y, 1)
+        y = self.linear(y)
+        return y
+
+model = Model()
+inputs = (torch.randn(1,1,16,16),)
+outputs = model(*inputs)
+print(f"Model output: {outputs}")
+```
+
+### Supporting Varying Input Sizes (Dynamic Shapes)
+
+The PyTorch export process uses the example inputs provided to trace through the model and reason about the size and type of tensors at each step. Unless told otherwise, export will assume a fixed input size equal to the example inputs and will use this information to optimize the model.
+
+Many models require support for varying input sizes. To support this, export takes a `dynamic_shapes` parameter, which informs the compiler of which dimensions can vary and their bounds. This takes the form of a nested dictionary, where keys correspond to input names and values specify the bounds for each input.
+
+In the example model, inputs are provided as 4-dimensions tensors following the standard convention of batch, channels, height, and width (NCHW). An input with the shape `[1, 3, 16, 16]` indicates 1 batch, 3 channels, and a height and width of 16.
+
+Suppose your model supports images with sizes between 16x16 and 1024x1024. The shape bounds can be specified as follows:
+
+```
+dynamic_shapes = {
+    "x": {
+        2: Dim("h", min=16, max=1024),
+        3: Dim("w", min=16, max=1024),
+    }
+}
+
+ep = torch.export.export(model, inputs, dynamic_shapes=dynamic_shapes)
+```
+
+In the above example, `"x"` corresponds to the parameter name in `Model.forward`. The 2 and 3 keys correpond to dimensions 2 and 3, which are height and width. As there are no specifications for batch and channel dimensions, these values are fixed according to the example inputs.
+
+ExecuTorch uses the shape bounds both to optimize the model and to plan memory for model execution. For this reason, it is advised to set the dimension upper bounds to no higher than needed, as higher bounds increase memory consumption.
+
+For more complex use cases, dynamic shape specification allows for mathematical relationships between dimensions. For more information on dynamic shape specification, see [Expressing Dynamism](https://pytorch.org/docs/stable/export.html#expressing-dynamism).
+
+## Export and Lowering
```

Comment:
THoughts on swapping "Export and Lowering" and "Supporting Varying Input Sizes (Dynamic Shapes)" sections.

Reply:
Talk about the simple case, and expand into complicated dynamic shapes if necessary.